mempool/reactor.go

```diff
@@ -93,7 +99,18 @@ func (memR *Reactor) GetChannels() []*p2p.ChannelDescriptor {
 // It starts a broadcast routine ensuring all txs are forwarded to the given peer.
 func (memR *Reactor) AddPeer(peer p2p.Peer) {
 	if memR.config.Broadcast {
-		go memR.broadcastTxRoutine(peer)
+		go func() {
+			if memR.config.ExperimentalMaxUsedOutboundPeers > 0 {
+				if err := memR.activeConnectionsSemaphore.Acquire(context.TODO(), 1); err != nil {
+					memR.Logger.Error("Failed to acquire semaphore: %v", err)
```

Comment:
Is this expected to happen in normal conditions? If not, consider a panic

Reply:
Should only happen if the context expires, which should not happen here.

Reply:
Should we really consider this behaviour that shouldn't happen? I think it just causes confusion on readers, as the error might be interpreted as a failure to acquire (immediately). Moreover, context.TODO() in production code is not great.